components/jstraceable_derive/lib.rs

```diff
@@ -149,21 +149,15 @@ fn js_traceable_derive(s: synstructure::Structure) -> proc_macro2::TokenStream {
     let mut asserts = quote!();
     let match_body = s.each(|binding| {
         for attr in binding.ast().attrs.iter() {
-            match attr.parse_meta().unwrap() {
-                syn::Meta::Path(ref path) | syn::Meta::List(syn::MetaList { ref path, .. }) => {
-                    if path.is_ident("no_trace") {
-                        asserts.extend(assert_not_impl_traceable(&binding.ast().ty));
-                        return None;
-                    } else if path.is_ident("custom_trace") {
-                        return Some(quote!(<crate::dom::bindings::trace::CustomTraceable>::trace(#binding, tracer);));
-                    }
-                },
-                syn::Meta::NameValue(syn::MetaNameValue { ref path, .. }) => {
-                    // if reason provided we can skip JSTraceable check
-                    if path.is_ident("no_trace") {
-                        return None;
-                    }
-                },
+            if attr.path().is_ident("no_trace") {
+                // if no reason is provided to `no_trace`
```

Comment:
Do you mind included the slightly more verbose comment here. I had to do a little bit of hunting to understand this code.

```rust
 // Do a `JSTraceable` check unless a reason is provided with `no_check`.
```

Reply:
Actually I spit this comment in two (se next commented line) so it makes more sense as a whole then separate. There is no `no_check` what I wanted to say is that you can put `#[no_trace]` and this will make sure that the type of the field is not JSTraceble, or if you put `#[no_trace="I know this ..."]` then we will not do such check.